### **Importacion de Librerias**

In [2]:
import pandas as pd

### **Conversion de csv a Df**

In [8]:
df = pd.read_csv('data/institutions.csv', encoding='latin1')
df.head()


Unnamed: 0         CODIGO DISTRITO DEPARTAMENTO   MUNICIPIO  \
0           0  12-01-0001-42   12-002   SAN MARCOS  SAN MARCOS   
1           1  12-01-0002-42   12-002   SAN MARCOS  SAN MARCOS   
2           2  12-01-0003-42   12-100   SAN MARCOS  SAN MARCOS   
3           3  12-01-0004-42   12-002   SAN MARCOS  SAN MARCOS   
4           4  12-01-0006-42   12-002   SAN MARCOS  SAN MARCOS   

                            ESTABLECIMIENTO  \
0           EODP NO. 1 "FRANCISCA SANDOVAL"   
1    EODP NO. 2 'GEORGINA RODAS DE ESCOBAR'   
2  COLEGIO PARTICULAR MIXTO SAGRADO CORAZON   
3         COLEGIO EVANGELICO MIXTO NAZARENO   
4                      COLEGIO "SAN MARCOS"   

                                           DIRECCION  TELEFONO  \
0                            7A. AVENIDA 5-74 ZONA 2  55315928   
1  AVE. EL REFORMADOR COLONIA JUSTO RUFINO BARRIO...  42914527   
2                           10A. AVENIDA 9-71 ZONA 3  77601164   
3                             15 AVENIDA 5-36 ZONA 5  44656690   
4                           6A. AVENIDA, 4-92 ZONA 2  77604147   

                           SUPERVISOR  \
0  AMILSA YAMILETH ESTRADA RODRÍGUEZ   
1  AMILSA YAMILETH ESTRADA RODRÍGUEZ   
2      JORGE EBERTO SOLORZANO SANCHEZ   
3  AMILSA YAMILETH ESTRADA RODRÍGUEZ   
4  AMILSA YAMILETH ESTRADA RODRÍGUEZ   

                                   DIRECTOR     NIVEL   SECTOR    AREA  \
0           ROXANA ELIZABETH LÓPEZ FUENTES  PARVULOS  OFICIAL  URBANA   
1          EMILIA MAXIMINA CARDONA BAUTISTA  PARVULOS  OFICIAL  URBANA   
2       MAGNOLIA ROSALETT CIFUENTES BARRIOS  PARVULOS  PRIVADO  URBANA   
3         ENNA DEL CARMEN GIRÓN RODRÍGUEZ  PARVULOS  PRIVADO  URBANA   
4  SOR CAROLINA PATRICIA MARTÍNEZ TRUJILLO  PARVULOS  PRIVADO  URBANA   

    STATUS   MODALIDAD   JORNADA             PLAN DEPARTAMENTAL  
0  ABIERTA  MONOLINGUE  MATUTINA  DIARIO(REGULAR)    SAN MARCOS  
1  ABIERTA  MONOLINGUE  MATUTINA  DIARIO(REGULAR)    SAN MARCOS  
2  ABIERTA  MONOLINGUE  MATUTINA  DIARIO(REGULAR)    SAN MARCOS  
3  ABIERTA  MONOLINGUE  MATUTINA  DIARIO(REGULAR)    SAN MARCOS  
4  ABIERTA  MONOLINGUE  MATUTINA  DIARIO(REGULAR)    SAN MARCOS

In [9]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 56457 entries, 0 to 56456
Data columns (total 18 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   Unnamed: 0       56457 non-null  int64 
 1   CODIGO           56457 non-null  object
 2   DISTRITO         56457 non-null  object
 3   DEPARTAMENTO     56457 non-null  object
 4   MUNICIPIO        56457 non-null  object
 5   ESTABLECIMIENTO  56457 non-null  object
 6   DIRECCION        56457 non-null  object
 7   TELEFONO         56457 non-null  object
 8   SUPERVISOR       56457 non-null  object
 9   DIRECTOR         56457 non-null  object
 10  NIVEL            56457 non-null  object
 11  SECTOR           56457 non-null  object
 12  AREA             56457 non-null  object
 13  STATUS           56457 non-null  object
 14  MODALIDAD        56457 non-null  object
 15  JORNADA          56457 non-null  object
 16  PLAN             56457 non-null  object
 17  DEPARTAMENTAL    56457 non-null

### Descripción del Dataset:

- **Número de filas (datos crudos):** 56,457  
- **Número de variables (columnas):** 18

### Nombres de las variables:
1. `Unnamed: 0` (El indice original del archivo al momento de hacer el scrapping)
2. `CODIGO`
3. `DISTRITO`
4. `DEPARTAMENTO`
5. `MUNICIPIO`
6. `ESTABLECIMIENTO`
7. `DIRECCION`
8. `TELEFONO`
9. `SUPERVISOR`
10. `DIRECTOR`
11. `NIVEL`
12. `SECTOR`
13. `AREA`
14. `STATUS`
15. `MODALIDAD`
16. `JORNADA`
17. `PLAN`
18. `DEPARTAMENTAL`


### **Estrategia de Limpieza Del Conjunto de datos**
1. Para la variable `Unnamed: 0` se planea eliminar completamente ya que este es el índice creado al momento de hacer webscraping y no aporta valor analítico al conjunto de datos. Su presencia puede generar redundancia o confusión, ya que pandas asigna automáticamente un índice al importar el archivo.
2. Para la variable `NIVEL`, se eliminarán todas las filas cuyo valor corresponda a niveles educativos superiores al diversificado, ya que el enfoque del análisis estará centrado únicamente en la educación preprimaria, primaria, básica y diversificada.
3. Para el tratamiento de los tipos de datos, se realizará una conversión (cast) explícita:
    - Las variable `TELEFONO` sera convertidas a tipo `int`
    - Todas las demás variables serán convertidas a tipo `string`, ya que representan categorías, nombres o descripciones textuales
    - Las variable `CODIGO` y `DISTRITO` se convertiran a `string`, para mantener su formato
4. En todas las variables Eliminar posibles espacios extra al inicio o al final que puedan afectar comparaciones.


